### PR TITLE
Enable MLKit 7 [Sept 2024] for Barcode Scanning

### PR DESCRIPTION
**Pull request by bengavin**
Updates to pull in the latest Sept 2024 release of MLKit for Barcode Scanning

### CODE REVIEW

File: source/MLKit/Core/Enums.cs [hunk 1]
```diff
@@ -4,8 +4,15 @@
 namespace MLKit.Core {
 	[Native]
 	public enum ImageSourceType : long {
-		ImageSourceTypeImage = 0,
-		ImageSourceTypePixelBuffer = 1,
-		ImageSourceTypeSampleBuffer = 2
+		Image = 0,
```
Comment (AdamEssenmacher) [inline source/MLKit/Core/Enums.cs:7]: I don't think we should be renaming these. Aside from it being a breaking change, the original names are accurate to the ObjectiveC naming

File: source/MLKit/Core/Enums.cs [hunk 2]
```diff
@@ -4,8 +4,15 @@
...
+		PixelBuffer = 1,
+		SampleBuffer = 2
+	}
+
+	[Native]
+	public enum ModelDownloadUserInfoKey : long
```
Comment (AdamEssenmacher) [inline source/MLKit/Core/Enums.cs:13]: I don't understand this. It doesn't look like it should be an enum. It's a constant NSString, and so should be defined as such in ApiDefinition.cs, right?

I'm assuming you added it for a reason though. Can you add some context?

**Comment by AdamEssenmacher** [comment]
The [iOS MLKit SDK release notes](https://developers.google.com/ml-kit/release-notes#september_30_2024) seem to indicate that the latest release is just bumping dependency versions.

I diffed the header files for MLImage, MLKitBarcodeScanning, MLKitCommon, and MLKitVision. It looks like they are unchanged, which means the API surface has not changed, which means there should not be any need to update `ApiDefinition or `Enums`.

File: source/MLKit/Core/Enums.cs (hunk 1 above)
Comment (bengavin) [reply source/MLKit/Core/Enums.cs:7]: Yeah, this one is on me, I was experimenting with Copilot assisting with validation that things matched the headers and it suggested I make the switch, never went back and verified against the header itself.

File: source/MLKit/Core/Enums.cs (hunk 2 above)
Comment (bengavin) [reply source/MLKit/Core/Enums.cs:13]: Same as above, missed this one in my cleanup

Comment (bengavin) [reply source/MLKit/Core/Enums.cs]: alright, I think I've updated this to match the pattern I've seen in other libraries in the repo and it should accurately reflect the underlying header files.

**Comment by bengavin** [comment]
@AdamEssenmacher - I've added the nanopb and PromisesObjC nuget packages here, let me know if you'd like any changes to how I've laid those out.  I'll probably look to convert the other shared libraries that MLKit requires with a similar pattern if we're ok with this approach.  Given the 'partial' nature of the conversion, I haven't yet gotten to the next level of integration to verify that this approach works properly, just wanted to check in that there aren't any obvious pieces missing before I convert the other items.

**Comment by AdamEssenmacher** [comment]
I appreciate your efforts on this. There were a handful of small issues, but rather than go back and forth I just made a new PR and manually merged stuff in.

Key notes:

- To separate MLKit off of Firebase.Core entirely, five new packages/artifact configurations needed to be added (including nanopb and GoogleDataTransport)
- I separated the new dependency nugets for the five Google packages out into their own PRs.
- The new package dependencies are set up to use package references with version constraints mirroring their associated pods.
- Rather than introduce a 'shared' area, I just kept the new packages under Google (because they are owned by Google)
- I left currently unsupported MLKit artifact configurations alone.
- GoogleUtilitiesComponents could be removed entirely.
- API definitions unrelated to the upgrade were left for future, focused PRs
- Timestamp members were not renamed.
- SupportedOSPlatformVersions were bumped to 13 for Firebase and 16 for MLKit
- Firebase.Auth and Firebase.RemoteConfig had some minor nullability updates associated with the version upgrade.

**Pull request closed without merging**